### DISCUSSION TITLE
Cannot run frontend and backend by following the steps from the Docs

### DISCUSSION THREAD

**Discussion by a-imran99** [highlight/highlight#9645] [General]
Cannot run frontend and backend by following the steps from the Docs
Some packages in Backend seems to be moved to new location and frontend is not hosted on localhost:3000

Help please! @Vadman97

**Comment by manimovassagh** [comment]
The self-hosted dev setup docs can get stale quickly as the monorepo evolves. Here's what typically trips people up and how to get past it:

**Backend:**

The Go backend dependencies have moved around over time. Make sure you're on the right Go version (check `go.mod` in the repo root) and run:
```bash
cd backend
go mod download
make start-dev
```

If packages are missing or relocated, try:
```bash
go mod tidy
```

The backend also depends on several infrastructure services (ClickHouse, Kafka, PostgreSQL, Redis, OpenSearch). The easiest way to get those running is via the Docker Compose file in the `docker/` directory:
```bash
docker compose -f docker/compose.yml up -d
```
That spins up all the backing services without running the Highlight app itself.

**Frontend:**

The frontend hasn't been on `localhost:3000` for a while. It now runs on `localhost:3001` (or sometimes `localhost:8080` depending on the configuration). Check `frontend/package.json` for the actual dev server port.

```bash
cd frontend
yarn install
yarn start
```

If `yarn` fails, check that you're using the version specified in the root `package.json` `engines` field or `.tool-versions` file.

**Quickest path:**

If you just want to get the full stack running without debugging build issues, the hobby/self-hosted Docker setup is more reliable:
```bash
git clone https://github.com/highlight/highlight.git
cd highlight/docker
./run-hobby.sh
```
This pulls pre-built images and wires everything together.

What specific errors are you hitting? If you share the actual error output, it'll be easier to pinpoint the issue.
